bundles/io/org.eclipse.smarthome.io.net/src/main/java/org/eclipse/smarthome/io/net/http/internal/NamedThreadFactory.java

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.io.net.http.internal;
+
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Thread factory class to create custom named threads
+ *
+ * @author Michael Bock - initial API
+ */
+public class NamedThreadFactory implements ThreadFactory {
```

Comment:
We already have such a factory in `ThreadPoolManager.NamedThreadFactory` - might it make sense to move that out there to a public class and reuse it here?

Reply:
This implementation is not complete. I will try to implement a solution which names the threads based on the actual consumer. If this is not realizable, we can get rid completely of the class.

Reply:
By now it is complete and I would say that this code can actually be removed and you should make use of the ESH [ThreadPoolManager](https://github.com/eclipse/smarthome/blob/master/bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/common/ThreadPoolManager.java) instead.